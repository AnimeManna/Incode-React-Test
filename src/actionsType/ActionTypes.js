export const XHR_CLIENTS_START = 'XHR_CLIENTS_START'
export const XHR_CLIENTS_SUCCESS = 'XHR_CLIENTS_SUCCESS'
export const XHR_CLIENTS_ERROR = 'XHR_CLIENTS_ERROR'