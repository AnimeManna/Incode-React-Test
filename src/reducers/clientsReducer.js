import {
    XHR_CLIENTS_SUCCESS,
    GET_CLIENT,
    SEARCH_CLIENT
} from "../ActionsType/ActionTypes";


const initialState={
    clients:[

    ],
    user:{

    },
    viewClients:[

    ]
}


export default (state=initialState, action) =>{
    switch (action.type){
        case XHR_CLIENTS_SUCCESS:{
            return{
                ...state,
                clients:[
                    ...action.payload
                ],
                viewClients:[
                    ...action.payload
                ]
            }
        }
        case GET_CLIENT:{
            console.log('REDUUUUCER',action.payload);
            let someClient = state.clients.find((value)=>{
                return (value.id===action.payload)
            })
            console.log('SOME CLIENTS',someClient)
            return{
                ...state,
                user:{
                    ...someClient
                }
            }
        }
        case SEARCH_CLIENT:{
            const {payload} = action
           const newClients =  state.clients.filter((value)=>{
               for(let key in value){
                   for(let newKey in value[key]){
                       if(value[key][newKey].includes(payload)){
                           return value
                           continue;
                       }
                   }
               }
            })
            return {
                ...state,
                viewClients: newClients
            }
        }
        default:{
            return state
        }
    }
}