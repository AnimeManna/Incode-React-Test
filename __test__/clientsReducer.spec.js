import clientsReducer from '../src/reducers/clientsReducer'

import {XHR_CLIENTS_SUCCESS} from "../src/actionsType/ActionTypes";

describe('clientsReducer', ()=>{
    describe('InitalState',()=>{
        test('isCorrect',()=>{
            const action = {}
            const initialState={
                user:{},
                clients:[],
                viewClients:[]
            }
            expect(clientsReducer(undefined, action)).toEqual(initialState)
        })
    })

    describe('XHR_CLIENTS_SUCCEESS',()=>{
        test('returns the correct state',()=>{
            const action = {type:XHR_CLIENTS_SUCCESS, payload:[{test:'test'}]}
            expect(clientsReducer(undefined, action)).toMatchSnapshot()
        })
    })
})