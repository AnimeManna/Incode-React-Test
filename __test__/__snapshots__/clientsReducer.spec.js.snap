// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clientsReducer XHR_CLIENTS_SUCCEESS returns the correct state 1`] = `
Object {
  "clients": Array [
    Object {
      "test": "test",
    },
  ],
  "user": Object {},
  "viewClients": Array [
    Object {
      "test": "test",
    },
  ],
}
`;
